parsec/core/gui/app.py

```diff
@@ -80,7 +84,31 @@ def _before_quit():
                 continue
 
 
-def run_gui(config: CoreConfig, start_arg: str = None):
+@contextmanager
+def fail_on_first_exception(diagnose, kill_window):
```

Comment:
Wouldn't it be more explanatory to call this something like `diagnose_exceptions_catcher`, then have a if in `run_gui`:
```python
if diagnose:
    with diagnose_exceptions_catcher(kill_window):
        return app.exec_()
else:
    return app.exec_()
```

Reply:
:+1: